### ISSUE TITLE
"Undefined index: type" when trying to save page

### ISSUE THREAD

**Issue by teevik** [paulmassen/grav-plugin-seo#34]
Not sure why this error happens, but when i disable the SEO plugin the error doesn't happen

![image](https://user-images.githubusercontent.com/9365365/33835926-3de69d34-de88-11e7-9b0a-c1ca1d91d9a3.png)

**Comment by paulmassen** [comment]
Ah, someone else reported this issue.
Does it happen on every page ?
I suspect the "required: true" attribute to be at fault.
Can you try removing this line here: https://github.com/paulmassen/grav-plugin-seo/blob/master/blueprints/seo.yaml#L198

Maybe it will fix the problem

**Comment by paulmassen** [comment]
Another possibility, is that the blueprint from the plugin has trouble being added to your custom blueprint.

It would help if you could paste your blueprint here.

Thanks!

**Comment by teevik** [comment]
Yeah removing the `required: false` didn't do anything, but here is my blueprints: 
`default.yaml`
```yaml
form:
  fields:
    tabs:
      type: tabs
      active: 1

      fields:
        content:
          type: tab
          title: Content
          fields:
            header.title:
              type: text
              label: Page Title
            header.background:
              type: file
              label: Page Background
              destination: "@self"
              accept: 
                - "image/*"
                - "video/*"
              
```
`home.yaml`
```yaml
extends@: default

form:
  fields:
    tabs:
      type: tabs
      active: 1

      fields:
        content:
          type: tab
          title: Content

          fields:
            header.hero_header_text:
              type: text
              label: Hero header text
            header.hero_sub_text:
              type: editor
              label: Hero sub text
```

**Comment by paulmassen** [comment]
hmm, just tried with your blueprint, and I can't replicate the problem.
I guess you are running latest grav and admin?

Would it be possible for you to zip and send me your `user/` folder?

**Comment by teevik** [comment]
Yep, here you go
[user.zip](https://github.com/paulmassen/grav-plugin-seo/files/1548539/user.zip)

**Comment by paulmassen** [comment]
Thanks, it will be helpful!

I'm taking a look right now but it seems like I'm still not able to replicate this. From your screenshot, it looks like the problem happens on Home page, but there is no such issue on my install.

Are you running it locally? php7 or php 5+ ? Any more info would be welcomed!

**Comment by teevik** [comment]
Huh alright, will check it out and see if i can figure out what's happening. I'm using php 7

**Comment by renards** [comment]
I could replicate the error :

* If I delete the `home.yaml` file the error disappears.
or
* If I change this part of the `seo.yaml` file (line 7 - 12) from 
```
      fields:
        content:
          fields:
            content:
              id: content
              ordering@: header.mykeyword
```
to
```
      fields:
        content:
          fields:
            whatever:
              id: whatever
              ordering@: header.mykeyword
```
saving works.

Hope this helps a bit.

**Comment by teevik** [comment]
Oh i see, thank you so much!

**Comment by renards** [comment]
You're welcome :slightly_smiling_face: 

I did not test the second solution in detail. I think @paulmassen can say if it has a deeper impact.

**Comment by paulmassen** [comment]
Good catch @renards ! 

This was indeed a problem with the blueprint on the seo plugin, on pages that do not use the default editor field.
The blueprints target the content field  in order to add an id. When there is no content field, it causes some problem apparently. I removed it, as this is only for another feature i'm working on in another branch.

I released the 2.1.6  which should solve this problem all along with some small  bugfixes.

Thanks again, 

Paul
